kubectl-cloudflow/deploy/deploy.go

```diff
@@ -92,6 +95,311 @@ func GetCloudflowApplicationDescriptorFromDockerImage(dockerRegistryURL string,
 	return spec, *pulledImage
 }
 
+// GetCloudflowApplicationSpecFromDockerImages pulls a image and extracts the Cloudflow Application descriptor from a docker label
+func GetCloudflowApplicationSpecFromDockerImages(imageRefs []ImageReference,
+	blueprintConfig *configuration.Config, replicas map[string]int) (domain.CloudflowApplicationSpec, []*docker.PulledImage) {
+
+	apiversion, apierr := exec.Command("docker", "version", "--format", "'{{.Server.APIVersion}}'").Output()
+	if apierr != nil {
+		util.LogAndExit("Could not get docker API version, is the docker daemon running? API error: %s", apierr.Error())
+	}
+
+	trimmedapiversion := strings.Trim(string(apiversion), "\t \n\r'")
+	client, error := docker.GetClient(trimmedapiversion)
+	if error != nil {
+		client, error = docker.GetClient("1.39")
+		if error != nil {
+			fmt.Printf("No compatible version of the Docker server API found, tried version %s and 1.39", trimmedapiversion)
+			panic(error)
+		}
+	}
+
+	// get all streamlet descriptors, image digests and pulled images in arrays
+	var streamletDescriptors []domain.Descriptor
+	var imageDigests []string
+	var apiVersion string
+	var pulledImages []*docker.PulledImage
+	var deployImages = make(map[string]string)
+	for _, imageRef := range imageRefs {
+		streamletsDescriptorsDigestPair, version := docker.GetCloudflowStreamletDescriptorsForImage(client, imageRef.FullURI)
```

Comment:
I think in this naming we can drop the 'Cloudflow' bit.

Reply:
👍